src/lib_c/x86_64-windows-msvc/c/time.cr

```diff
@@ -0,0 +1,33 @@
+lib LibC
+  struct Filetime
+    low_date_time : DWord
+    high_date_time : DWord
+  end
+
+  struct SystemTime
+    year : Word
+    month : Word
+    day_of_week : Word
+    day : Word
+    hour : Word
+    minute : Word
+    second : Word
+    milliseconds : Word
+  end
+
+  struct TimeZoneInformation
+    bias : Long
+    standard_name : StaticArray(WChar, 32)
+    standard_date : SystemTime
+    standard_bias : Long
+    daylight_name : StaticArray(WChar, 32)
+    daylight_date : SystemTime
+    daylight_bias : Long
+  end
+
+  fun get_time_zone_information = GetTimeZoneInformation(tz_info : TimeZoneInformation*) : DWord
+  fun get_system_time_as_file_time = GetSystemTimeAsFileTime(time : Filetime*)
```

Comment:
Looks like you want [GetSystemTimePreciseAsFileTime](https://msdn.microsoft.com/en-us/library/windows/desktop/hh706895(v=vs.85).aspx) actually. It's fine that the precision is 100ns, it's fairly impossible to synchronise the system clock to be more accurate than that anyway.

Reply:
Hm, I don't know if it makes any difference. Both return a `FILETIME` structure with 100 ns resolution and the values returned by both functions seem to be equally precise (at least on my system). 
`GetSystemTimePreciseAsFileTime` is only available on Windows 8+. I don't know which windows versions Crystal should support but it should easily be older ones.

Reply:
I've taken a look at Go's implementation and they use `GetSystemTimePreciseAsFileTime` as well. Might need some more investigation about the actual difference.

Reply:
We should use the more precise one if it's there. The old one is remarkably inaccurate: see https://github.com/dotnet/coreclr/issues/5061.

Reply:
Should we have a fallback for API prior to Windows 8?

Reply:
Yes we should do the same library open and function pointer trick that the coreclr does in the pr linked from that issue.